src/sage/rings/polynomial/laurent_polynomial.pyx

```diff
@@ -1316,6 +1316,65 @@ cdef class LaurentPolynomial_univariate(LaurentPolynomial):
             return ~self
         raise ArithmeticError("element is not a unit")
 
+    def xgcd(self, other):
+        """
+        Extended `gcd` for univariate Laurent polynomial rings over a field.
+
+        EXAMPLES::
+
+            sage: S.<t> = LaurentPolynomialRing(QQ)
+            sage: (t^-2 + 1).xgcd(t^-3 + 1)
+            (1, 1/2*t^2 - 1/2*t^3 - 1/2*t^4, 1/2*t^3 + 1/2*t^4)
+        """
+        R = self.parent()
+        S = R.polynomial_ring()
+        f, df = self.monomial_reduction()
+        g, dg = other.monomial_reduction()
+        h, p, q = f.xgcd(g)
+        return R(h), p / df, q / dg
+
+    def inverse_mod(a, m):
+        """
+        Invert the polynomial ``a`` with respect to ``m``, or raise a :class:`ValueError`
+        if no such inverse exists.
+
+        The parameter ``m`` may be either a single polynomial or an ideal
+        (for consistency with :meth:`inverse_mod` in other rings).
+
+        EXAMPLES::
+
+            sage: S.<t> = LaurentPolynomialRing(QQ)
+            sage: f = inverse_mod(t^-2 + 1, t^-3 + 1); f
+            1/2*t^2 - 1/2*t^3 - 1/2*t^4
+            sage: f * (t^-2 + 1) + (1/2*t^4 + 1/2*t^3) * (t^-3 + 1)
+            1
+
+        ALGORITHM: Solve the system `as + mt = 1`, returning `s` as the inverse
+        of `a` mod `m`.
+        """
+        from sage.rings.ideal import is_Ideal
+        if is_Ideal(m):
+            v = m.gens_reduced()
+            if len(v) > 1:
+                raise NotImplementedError("Don't know how to invert modulo non-principal ideal %s" % m)
+            m = v[0]
+        if m.degree() == 1 and m[1].is_unit():
+            # a(x) mod (x-r) = a(r)
+            r = -m[0]
+            if not m[1].is_one():
+                r *= m.base_ring()(~m[1])
+            u = a(r)
+            if u.is_unit():
+                return a.parent()(~u)
+        if not a.parent().is_exact():
+            raise NotImplementedError("only implemented when the base ring is exact")
```

Comment:
I (still) don't see why you need this exactness check.

Reply:
Let us take it out